language/move-prover/bytecode/src/usage_analysis.rs

```diff
@@ -300,6 +301,37 @@ impl<'a> TransferFunctions for MemoryUsageAnalysis<'a> {
     }
 }
 
+impl<'a> MemoryUsageAnalysis<'a> {
+    /// Compute usage information for the given spec. This spec is injected in later
+    /// phases into the code, but we need to account for it's memory usage already here
+    /// as spec injection itself depends on this information.
+    fn compute_spec_usage(&self, spec: &Spec, state: &mut UsageState) {
+        use ConditionKind::*;
+        for cond in &spec.conditions {
+            let mut used_memory = cond.exp.used_memory(self.cache.global_env());
+            for exp in &cond.additional_exps {
+                used_memory.extend(exp.used_memory(self.cache.global_env()));
+            }
+            match &cond.kind {
+                Ensures | AbortsIf | Emits => {
+                    state.add_direct_accessed_iter(used_memory.into_iter().map(|(usage, _)| usage));
```

Comment:
@mengxu-fb notice I needed to use accessed memory here to get this work; assumed/asserted would not work because we do not use those in subsequent phases, e.g. when we generate entrypoint assumptions in spec_instrumentation. Perhaps assumed/asserted/modified should be a subset of accessed by construction?

Reply:
Currently `modified` is a subset of `accessed` by construction, and the other two are not. But I think they should be a subset of `accessed`.

Reply:
And in fact, even with the "mono at the end" work, we don't need to check `assumed` and `asserted` individually, so I'll suggest that we remove `assumed` and `asserted` and merge those into `accessed`. We can do it either in this PR or in a separate one.

Reply:
I've a hunch we may want to distinguish assumed/asserted in some future. I instead removed the `.all` field and let the `accesssed`  field be the one which accumulates all kind of usages,

Reply:
Sounds good to me